server/src/apub/fetcher.rs

```diff
@@ -43,36 +38,111 @@ use chrono::NaiveDateTime;
 static ACTOR_REFETCH_INTERVAL_SECONDS: i64 = 24 * 60 * 60;
 
 // Fetch nodeinfo metadata from a remote instance.
-fn _fetch_node_info(domain: &str) -> Result<NodeInfo, Error> {
+async fn _fetch_node_info(client: &Client, domain: &str) -> Result<NodeInfo, LemmyError> {
   let well_known_uri = Url::parse(&format!(
     "{}://{}/.well-known/nodeinfo",
     get_apub_protocol_string(),
     domain
   ))?;
-  let well_known = fetch_remote_object::<NodeInfoWellKnown>(&well_known_uri)?;
-  Ok(fetch_remote_object::<NodeInfo>(&well_known.links.href)?)
+
+  let well_known = fetch_remote_object::<NodeInfoWellKnown>(client, &well_known_uri).await?;
+  let nodeinfo = fetch_remote_object::<NodeInfo>(client, &well_known.links.href).await?;
+
+  Ok(nodeinfo)
+}
+
+enum RequestError {
+  Send(actix_web::client::SendRequestError),
+  Recv(awc::error::JsonPayloadError),
```

Comment:
apparently actix-web's JsonPayloadError is not the same as awc's JsonPayloadError